realm/realm-library/src/main/java/io/realm/internal/UncheckedRow.java

```diff
@@ -276,8 +287,15 @@ public boolean isNull(long columnIndex) {
         return nativeIsNull(nativePointer, columnIndex);
     }
 
+    /**
+     * Set null to a row pointer.
+     *
+     * @param columnIndex 0 based index value of the cell column.
+     */
     @Override
     public void setNull(long columnIndex) {
+        parent.checkImmutable();
+        getTable().checkNullValueIsLegal(columnIndex, getIndex());
```

Comment:
The nullability check is now moved to JNI. 

The impression I've had from other `UncheckedRow` functions in JNI is that they didn't seem to go out an extra length to validates items like they are checked in `CheckedRow`. (For the reason of speed, I suppose.) So nullability check was done at `CheckedRow` in Java.

After checking other functions in `UncheckedRow` JNI, I've finally noticed that `nativeSetString()` does check nullability and thus nullability check is now implemented in `UncheckedRow_setNull()`. 

Hope this addresses the issue properly.